### DISCUSSION TITLE
Show "Special style (N+1)" keymodes on o!m with exclusive tagging

### DISCUSSION THREAD

**Discussion by EdgyKing** [ppy/osu#28265] [Ideas]
Some maps on osu!mania use the "_Special style_" feature as a distinct playstyle for the keymode (_eg. 5k+1, 7k+1, 10k2s, etc_), but they are tagged as if they were a standard keymode (_eg. a 5k+1 map is autotagged as a usual 6k map_) both in-game and web. As they have a unusual key layout, the maps that use  "_Special style_" are very unlikely to their standard layout counterpart, being essencially different keymodes entirely and them being displayed as another keymode is misleading for players who doesn't know what is the "_Special style_" modes. Also, that's a feature "antimeta" players would benefit from, since those maps are already rare to show in the ranked section, being tagged as another keymode doesn't help for searching for "_Special style_" maps

As such, the maps with the "_Special style_" checkbox enabled with a single scratch table should be tagged their distinct keymode as follows:
- if a difficulty have 6 keys and the "_Special style_" enabled in the map's file, **it should be show as [5K+1]**
- if a difficulty have 8 keys and the "_Special style_" enabled in the map's file, **it should be show as [7K+1]**

For maps that utilize two scratch tables (_eg. 10k2s_) is a bit more complicated, but they should be fine to implement since those modes are already viable by the ranking criteria like this:
- if a difficulty have 6 keys, coop/dual-play **and** the "_Special style_" enabled, **it should be show as [10K+2]**
![image](https://github.com/ppy/osu/assets/40869733/0d66ca32-1255-4a12-8ab6-61205cc90186)
- _Exemple of how a 5k+1 [map](https://osu.ppy.sh/beatmapsets/2047832#mania/4275687) would show in web, instead of the current 6k._

And for the searching filters, adding new categories to compensate the new keymodes, eg:
- To filter maps that utilize the 7k+1, utilize `key=7+1`
- To filter maps that utilize the 10k+2, utilize `key=10+2`
It can be discussed if the old filters should work as well, since those modes still technically have more buttons than their usual keymode (_eg. 5k+1 being find with the `key=6` filter **in addition** with the `key=5+1` filter._)

**Comment by bunnerfly** [comment]
this would be so life changing please 🔥
